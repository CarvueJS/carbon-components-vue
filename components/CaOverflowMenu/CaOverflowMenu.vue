<template>
  <div data-overflow-menu tabindex="0" aria-label="Overflow menu description" class="bx--overflow-menu" >
    <svg class="bx--overflow-menu__icon" width="4" height="20" viewBox="0 0 4 20" fill-rule="evenodd">
      <circle cx="2" cy="2" r="2"></circle>
      <circle cx="2" cy="10" r="2"></circle>
      <circle cx="2" cy="18" r="2"></circle>
    </svg>
    <ul class="bx--overflow-menu-options" tabindex="-1">
      <slot></slot>
    </ul>
  </div>
</template>

<script>
import { OverflowMenu } from 'carbon-components';

export default {
  name: 'ca-overflow-menu',
  data: () => ({
    overflowMenuInstance: null,
  }),
  mounted() {
    this.overflowMenuInstance = OverflowMenu.create(this.$el);
  },
};
</script>

<style scoped>
.bx--overflow-menu-options:focus {
  outline: none;
}
</style>
