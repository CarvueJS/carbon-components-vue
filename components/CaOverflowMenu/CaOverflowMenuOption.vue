<template>
  <li class="bx--overflow-menu-options__option">
    <button class="bx--overflow-menu-options__btn" @click="onClick">
      <slot></slot>
    </button>
  </li>
</template>

<script>
export default {
  name: 'ca-overflow-menu-option',
  props: {
    onClick: {
      type: Function,
      default: () => {},
    },
  },
};
</script>
