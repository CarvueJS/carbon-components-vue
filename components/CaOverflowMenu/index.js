import CaOverflowMenu from './CaOverflowMenu';
import CaOverflowMenuOption from './CaOverflowMenuOption';

CaOverflowMenu.install = (Vue) => {
  Vue.component(CaOverflowMenu.name, CaOverflowMenu);
  Vue.component(CaOverflowMenuOption.name, CaOverflowMenuOption);
};

export default CaOverflowMenu;
