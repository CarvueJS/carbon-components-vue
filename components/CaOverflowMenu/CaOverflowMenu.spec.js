import { shallow } from 'vue-test-utils';
import CaOverflowMenu from './CaOverflowMenu.vue';
import CaOverflowMenuOption from './CaOverflowMenuOption.vue';

describe('CaOverflowMenu', () => {
  test('is a Vue instance', () => {
    const wrapper = shallow(CaOverflowMenu, { 
      slots: {
        default: CaOverflowMenuOption
      }
    });
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('Mathes snapshot', () => {
    const wrapper = shallow(CaOverflowMenu, { 
      slots: {
        default: CaOverflowMenuOption
      }
    });
    expect(wrapper.html()).toMatchSnapshot()
  });
})
