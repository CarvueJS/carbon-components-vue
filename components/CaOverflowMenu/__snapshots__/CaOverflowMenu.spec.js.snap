// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaOverflowMenu Mathes snapshot 1`] = `
<div data-overflow-menu="" tabindex="0" aria-label="Overflow menu description" class="bx--overflow-menu">
  <svg width="4" height="20" viewBox="0 0 4 20" fill-rule="evenodd" class="bx--overflow-menu__icon">
    <circle cx="2" cy="2" r="2"></circle>
    <circle cx="2" cy="10" r="2"></circle>
    <circle cx="2" cy="18" r="2"></circle>
  </svg>
  <ul tabindex="-1" class="bx--overflow-menu-options">
    <li class="bx--overflow-menu-options__option">
      <button class="bx--overflow-menu-options__btn"></button>
    </li>
  </ul>
</div>
`;
